libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -153,31 +153,36 @@ AC_WPNav::TerrainSource AC_WPNav::get_terrain_source() const
 ///     stopping_point_ne_cm should be the vehicle's stopping point (equal to the starting point of the next segment) if know or left as zero
 ///     should be called once before the waypoint controller is used but does not need to be called before subsequent updates to destination
 void AC_WPNav::wp_and_spline_init_cm(float speed_cms, Vector3f stopping_point_ne_cm)
+{    
+    wp_and_spline_init_m(speed_cms * 0.01, stopping_point_ne_cm * 0.01);
+}
+
+void AC_WPNav::wp_and_spline_init_m(float speed_ms, Vector3f stopping_point_ne_m)
 {    
     // check _wp_radius_cm is reasonable
     _wp_radius_cm.set_and_save_ifchanged(MAX(_wp_radius_cm, WPNAV_WP_RADIUS_MIN_CM));
 
     // check _wp_speed
-    _wp_speed_cms.set_and_save_ifchanged(MAX(_wp_speed_cms, WPNAV_WP_SPEED_MIN_CMS));
+    _wp_speed_cms.set_and_save_ifchanged(MAX(_wp_speed_cms, WPNAV_WP_SPEED_MIN_MS * 100.0));
 
     // initialise position controller
     _pos_control.init_U_controller_stopping_point();
     _pos_control.init_NE_controller_stopping_point();
 
     // initialize the desired wp speed
-    _check_wp_speed_change = !is_positive(speed_cms);
-    _wp_desired_speed_ne_cms = is_positive(speed_cms) ? speed_cms : _wp_speed_cms;
-    _wp_desired_speed_ne_cms = MAX(_wp_desired_speed_ne_cms, WPNAV_WP_SPEED_MIN_CMS);
+    _check_wp_speed_change = !is_positive(speed_ms);
+    _wp_desired_speed_ne_ms = is_positive(speed_ms) ? speed_ms : get_default_speed_NE_ms();
+    _wp_desired_speed_ne_ms = MAX(_wp_desired_speed_ne_ms, WPNAV_WP_SPEED_MIN_MS);
 
     // initialise position controller speed and acceleration
-    _pos_control.set_max_speed_accel_NE_cm(_wp_desired_speed_ne_cms, get_wp_acceleration_cmss());
-    _pos_control.set_correction_speed_accel_NE_cm(_wp_desired_speed_ne_cms, get_wp_acceleration_cmss());
-    _pos_control.set_max_speed_accel_U_cm(-get_default_speed_down_cms(), _wp_speed_up_cms, _wp_accel_z_cmss);
-    _pos_control.set_correction_speed_accel_U_cmss(-get_default_speed_down_cms(), _wp_speed_up_cms, _wp_accel_z_cmss);
+    _pos_control.set_max_speed_accel_NE_m(_wp_desired_speed_ne_ms, get_wp_acceleration_mss());
+    _pos_control.set_correction_speed_accel_NE_m(_wp_desired_speed_ne_ms, get_wp_acceleration_mss());
+    _pos_control.set_max_speed_accel_U_m(-get_default_speed_down_ms(), get_default_speed_up_ms(), get_accel_U_mss());
+    _pos_control.set_correction_speed_accel_U_mss(-get_default_speed_down_ms(), get_default_speed_up_ms(), get_accel_U_mss());
 
     // calculate scurve jerk and jerk time
     if (!is_positive(_wp_jerk_msss)) {
-        _wp_jerk_msss.set(get_wp_acceleration_cmss());
+        _wp_jerk_msss.set(get_wp_acceleration_mss() * 100.0);
```

Comment:
the original line seems odd actually in that we appeared to be setting the _mss parameter with a _cmss value.  I think we should lose this, "*100" to fix the existing bug